ppb_vector/vector2.py

```diff
@@ -109,32 +186,32 @@ def __getitem__(self, item: typing.Union[str, int]) -> Real:
         else:
             raise TypeError
 
-    def __repr__(self) -> str:
+    def __repr__(self: VectorOrSub) -> str:
         return f"{type(self).__name__}({self.x}, {self.y})"
 
-    def __eq__(self, other: VectorLike) -> bool:
+    def __eq__(self: VectorOrSub, other: typing.Any) -> bool:
         if is_vector_like(other):
-            other = _mkvector(other)
+            other = Vector2.convert(other)
             return self.x == other.x and self.y == other.y
         else:
             return False
 
-    def __ne__(self, other: VectorLike) -> bool:
+    def __ne__(self: VectorOrSub, other: typing.Any) -> bool:
```

Comment:
Can't we replace the whole function with `not self == other`?

Reply:
Probably.